### PR TITLE
config: rename `ports.grpc` to `ports.xds`

### PR DESCRIPTION
**Pull request by dnephin**
Now that we have gRPC endpoints (streaming subscribe, and a new one coming in enterprise) that use the regular RPC port it seems likely that we will continue to use the RPC port for any "regular" RPC endpoints.

However, our config has a `ports.grpc` and `addresses.grpc` that apply only to the xDS interface. Since the consumer of the xDS interface is quite different from the RPC interface (expected to be localhost, and a Envoy proxy), it seems likely that this port will only ever be used for xDS, and all other things will continue to use the RPC port.

To make our config and documentation less confusing I've renamed this config setting to be `xds` instead of `grpc`. This change should be entirely backwards compatible. The old config setting and command line flag both continue to work, and the `consul connect envoy` command is still able to query for this port with the new name.

TODO:
* [x] the v2compat integration tests are failing because they use an older CLI with a newer agent. Need to restore the GRPC.Ports to the DebugConfig to keep that working.
* [x] changelog entry

### CODE REVIEW

File: command/agent/agent.go
Comment (rboyer) [inline command/agent/agent.go:203]:
```suggestion
	ui.Info(fmt.Sprintf("   Client Addr: %v (HTTP: %d, HTTPS: %d, xDS: %d, DNS: %d)", config.ClientAddrs,
```

File: agent/config/runtime.go
Comment (rboyer) [inline agent/config/runtime.go:722]:
```suggestion
	// hcl: ports { xds = int }
	// flags: -xds-port int
```

File: agent/config/runtime.go
Comment (rboyer) [inline agent/config/runtime.go:737]:
```suggestion
	// hcl: client_addr = string addresses { xds = string } ports { xds = int }
```

**Comment by dnephin** [comment]
Test failures is a known flake and unrelated to this change.

File: website/content/docs/agent/options.mdx
Comment (rboyer) [inline website/content/docs/agent/options.mdx:245]: Since our docs aren't versioned yet, I think we need to keep the two old grpc lines in here, but mark them as deprecated and link them to the new names.

**Review by rboyer — APPROVED** [review]
LGTM pending the one minor suggestion about retaining the old config keys on the website.

File: website/content/docs/agent/options.mdx
Comment (dnephin) [reply website/content/docs/agent/options.mdx:245]: Unfortunate, but makes sense

Comment (rboyer) [reply website/content/docs/agent/options.mdx:245]: Usually when we deprecate stuff like this we also indicate which version of consul it was deprecated in so that folks using older versions can still figure out what's going on for their version.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/408416.

File: website/content/docs/agent/options.mdx
Comment (dnephin) [reply website/content/docs/agent/options.mdx:245]: Ah, sorry I missed this comment. I wasn't sure which version this would land in yet. I guess it will be 1.11, so I can add that.

Comment (dnephin) [reply website/content/docs/agent/options.mdx:245]: https://github.com/hashicorp/consul/pull/10618

**Pull request merged**
